### DISCUSSION TITLE
Add mail template translations

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#3115] [Feature Requests]
Emails should be sent in the users preferred language

**Comment by andriusign** [comment]
I could contribute to this by translating Directus email templates (invite, pass reset, etc.) to Lithuanian and Russian languages. What would be the process? Templates could be selected by user's language property.

**Reply by rijkvanzanten** [reply]
> What would be the process?

Good question. I think we should setup a similar translations file on the API side for these emails, and then re-use those static strings inside the same email template. I wouldn't want to recreate the whole email template just for the new text, as that would make it pretty annoying to maintain if we ever need to change some markup.

I think the best way forward is to mirror the app's lang folder in the API, and also connect it up to Crowdin for translations

> Templates could be selected by user's language property.

This makes most sense to me 👍🏻

**Reply by rijkvanzanten** [reply]
We should also figure out if/how to allow custom translated strings to work in mail template extensions 🤔

**Comment by dannycoulombe** [comment]
I need this and willing to do a PR. Just to be sure, @rijkvanzanten, what would be your strategy at high-level if you'd have to do it?

**Reply by dannycoulombe** [reply]
I suggest the following, but I'm sure you know much better:

- passing a "x-locale" header in all API queries
- implementing a function in Directus SDK to update x-locale header
- creating an i18n service in Directus API
- export it in "services" object so we can use it from custom hooks and endpoints and pass it to mailService templates.

**Reply by rijkvanzanten** [reply]
> what would be your strategy at high-level if you'd have to do it?

Very good question! I think I would steer clear from a new header, as we generally do everything in the API through query parameters. As a matter of fact, for the system emails, we know what user we're sending the email too, so we might be able to pull the preferred language straight from the user. 🤔 

As for the translations themselves, I think we'll have to setup a similar translations structure as we have in the app, and hook it up to Crowdin (locales.directus.io) as well, so we can have proper community driven translations

**Reply by dannycoulombe** [reply]
That makes sense and we can avoid a new header. On the other end, we may not know the user's locale until he creates an account. Let's say I create a table newsletter where you could register to without having an account. We also need a way to force a language if not available from a user preference.

**Reply by rijkvanzanten** [reply]
> We also need a way to force a language if not available from a user preference.

True! Though could/would that be solved by having a configurable default language in directus_settings? 🤔

**Reply by dannycoulombe** [reply]
Yes and no.. If the user is navigating in Spanish, but the admin default language is set to English, how can he receives his confirmation email in Spanish otherwise?

**Reply by rijkvanzanten** [reply]
Ah yeah, good point 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
